### ISSUE TITLE
Android plugin fails to resolve some of the discovered services

### ISSUE THREAD

**Issue by egaga** [Nimrodda/flutter_nsd#7]
There's two bugs in the plugin:

First, you should not reuse the resolve listener. See https://stackoverflow.com/questions/25815162/listener-already-in-use-service-discovery

Secondly, and this is more problematic, having discovered multiple services, even if resolve listener is not shared, only one of the discovered services will be resolved. See: https://stackoverflow.com/questions/24665342/nsdmanager-doesnt-resolve-multiple-discovered-services

This can be reproduced by creating a couple of services with the same service type, 
e.g. in Macos `dns-sd -R Foo _http._tcp . 8899 mydevice=1` and `dns-sd -R Bar _http._tcp . 8899 mydevice=2` 

You can see that the problem goes away if you add some delay to before calling `nsdManager?.resolveService`, but of course that wouldn't be a proper fix.

This is a critical bug, and I would be grateful for fixing it!

**Comment by egaga** [comment]
Oh, it seems this has been fixed? https://github.com/Nimrodda/flutter_nsd/commit/7caf86a64d7289baba1daeedb132b28af1de301d#diff-d522d7763ab9c2fbd638d00d7042d518aa2f9a15e0a7ae3da32a9ca782b5baf9

But flutter package repository has old code!? Could you publish newer version there, please?

**Comment by Nimrodda** [comment]
I haven't had the chance yet to test the latest code in `master` so that's why I haven't published it. Have you tried cloning the repo and trying it locally to see if it fixes the issue? You can use a `path` instead of a version in pubspec.yaml to try it out.

```
dependencies:
  flutter_nsd: 
    path: /path/to/repo
```

**Comment by egaga** [comment]
Thanks, I tried it earlier quickly, but not comprehensively, just by replacing the plugin file locally in the flutter cache directory.
I didn't review the code properly but hopefully it works even when one of the resolving fails, so that the rest of the queue will be handled.

Great to know you can reference by repo, thanks!

**Comment by Nimrodda** [comment]
I did a brief test of `master` branch yesterday evening and found out that the discovered service is emitted twice in my case. Prior to #6 the same test resulted in _one_ emitted service. I'll have to take a deeper look either in the following evenings or in the weekend before I publish 1.2.0 to pub.dev.

Have you noticed the same issue on your end?

**Comment by egaga** [comment]
No, I just re-tested, and everything works, without duplicates.

Sure, you get duplicates if you press start / stop multiple times. e.g. If you press the start, and a little bit later again, you might see multiple values, but that's because the example code adds new values to the `services`.

In the listen callback, no duplicates, either.
Also if I register a new service after the app is already running, the listener finds it 👍🏼 .

I tested with Android Pixel 3a and also verified that iPhone works the same (both tested using usb-connection)
Of course, I have done this only a couple of times, so there could be some problem. Did you get a duplicate just by pressing start once and with a fresh build/run?

I ran `flutter clean` and build having this dependency definition for flutter_nsd:
```
flutter_nsd:
    git:
      url: git://github.com/Nimrodda/flutter_nsd.git
```
